### PR TITLE
[CodeQuality][Up_TO_PHP_81] Handle crash indentation on combination LevelSetList::UP_TO_PHP_81 with SetList::CODE_QUALITY

### LINKED ISSUE #7364: System error: \"PhpParser\\Internal\\TokenStream::getIndentationBefore(): Return value must be of type int, null returned
# Bug Report

| Subject        | Details   |
| :------------- | :---------------------------------------------------------------|
| Rector version | v0.13.10  |
| Installed as | composer dependency |

Hey guys!, I'm trying to upgrade and refactor a Legacy Code written in PHP 7.4 but I'm getting severals errors, this is one of them.

## Minimal PHP Code Causing Issue

for ($i = 0; $i < sizeof($psychiatry); $i++) {
	$psychiatry[$i]["sufficient_cancellation_notice"] = $s->isSufficientNoticeToCancel($psychiatry[$i]) ? 1 : 0;
}

for ($i = 0; $i < sizeof($psychotherapy); $i++) {
	$psychotherapy[$i]["sufficient_cancellation_notice"] = $s->isSufficientNoticeToCancel($psychotherapy[$i]) ? 1 : 0;
}

https://getrector.org/demo/168046d0-bb76-4c4b-b480-8a104e3e2ed9

## Expected Behaviour
Refactor the code to improve code quality and make compatible with PHP8.1

## Responsible Set of Rules
CODE_QUALITY

### PR DESCRIPTION
**Pull request by samsonasik**
Given the following code:

```php
for ($i = 0; $i < sizeof($psychiatry); $i++) {
	$psychiatry[$i]["sufficient_cancellation_notice"] = $s->isSufficientNoticeToCancel($psychiatry[$i]) ? 1 : 0;
}
```

It currently produce error:

```bash
There was 1 failure:

1) Rector\Core\Tests\Issues\IndentationCrash\IndentationCrashTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
assert($subStartPos >= 0 && $subEndPos >= 0) in /Users/samsonasik/www/rector-src/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php:608
```

Fixes https://github.com/rectorphp/rector/issues/7364

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @alejomoreira

**Comment by samsonasik** [comment]
it seems cause erorr in other tests

File: tests/Issues/IndentationCrash/config/configured_rule.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+use Rector\Config\RectorConfig;
+use Rector\Set\ValueObject\LevelSetList;
+use Rector\Set\ValueObject\SetList;
+
+return static function (RectorConfig $rectorConfig): void {
+    $rectorConfig->sets([LevelSetList::UP_TO_PHP_81, SetList::CODE_QUALITY]);
```
Comment (TomasVotruba) [inline tests/Issues/IndentationCrash/config/configured_rule.php]: What are the minimals 2 rules causing the crash?
There will be some invalid node replacement. Probably missing set original node null.

Comment (samsonasik) [reply tests/Issues/IndentationCrash/config/configured_rule.php]: change `sizeof` to `count` seems on of it:

https://github.com/rectorphp/rector-src/blob/bb68d8aa18c2b0ab3f5000c175fd6b81e9c0610a/config/set/code-quality.php#L89

other seems need to check one by one set list from php 5.4 to 8.1

Comment (samsonasik) [reply tests/Issues/IndentationCrash/config/configured_rule.php]: applied rules reduced to RenameFucntionRector, ForRepeatedCountToOwnVariableRector, then LevelSetList::UP_TO_PHP_81 https://github.com/rectorphp/rector-src/pull/2760/commits/b05501d199883b3d46dc189f01190cb3597411ad

Comment (samsonasik) [reply tests/Issues/IndentationCrash/config/configured_rule.php]: Reduced to only 4 rules:

```
Rector\CodeQuality\Rector\For_\ForRepeatedCountToOwnVariableRector;
Rector\Php71\Rector\FuncCall\CountOnNullRector;
Rector\Php81\Rector\FuncCall\NullToStrictStringFuncCallArgRector;
Rector\Renaming\Rector\FuncCall\RenameFunctionRector;
```

https://github.com/rectorphp/rector-src/pull/2760/commits/55efa059fc0a30377e0a9e40df943fee2e9acc4d

Comment (samsonasik) [reply tests/Issues/IndentationCrash/config/configured_rule.php]: https://getrector.org/demo/b3c0e6bc-073d-4cb1-a94e-37c2d06c64c1

**Comment by samsonasik** [comment]
Fixed 🎉

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**